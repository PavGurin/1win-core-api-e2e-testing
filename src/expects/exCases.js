import { expect } from 'chai';

export function checkInfoCase(data, idCases, cost, currency, id, max, min, name, priceMaxWin,
  priceMinWin, caseTypeId) {
  expect(data).to.be.an('array');
  expect(data[idCases].adjust).to.be.an('object');
  expect(data[idCases].cost).equal(cost);
  expect(data[idCases].currency).equal(currency);
  expect(data[idCases].id).equal(id);
  expect(data[idCases].max).equal(max);
  expect(data[idCases].min).equal(min);
  expect(data[idCases].name).equal(name);
  expect(data[idCases].priceMaxWin).equal(priceMaxWin);
  expect(data[idCases].priceMinWin).equal(priceMinWin);
  expect(data[idCases].caseTypeId).equal(caseTypeId);
}

export function checkCaseInfoByTypeId(data, idCases, caseTypeId, id) {
  expect(data[idCases].caseTypeId).equal(caseTypeId);
  expect(data[idCases].id).equal(id);
}

export function checkCaseResult(data, max, min) {
  expect(data.status).equal(200);
  expect(data.data.result).to.be.at.most(max);
  expect(data.data.result).to.be.at.least(min);
}
