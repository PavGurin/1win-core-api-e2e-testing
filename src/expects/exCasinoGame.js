export function checkGameInfo(receivedGame, expectedGame) {
  expect(receivedGame.id).toEqual(expectedGame.id);
  expect(receivedGame.nameEn).toEqual(expectedGame.nameEn);
  expect(receivedGame.nameRu).toEqual(expectedGame.nameRu);
  expect(receivedGame.hasDemo).toEqual(expectedGame.hasDemo);
  expect(receivedGame.imageUrl).toEqual(expectedGame.imageUrl);
}

export function checkGameArray(received, expected) {
  expect(received.length).toEqual(expected.length);

  expected.forEach((expectedGame) => {
    const receivedGame = received.find(game => game.id === expectedGame.id);
    expect(receivedGame.id).toEqual(expectedGame.id);
    expect(receivedGame.nameEn).toEqual(expectedGame.nameEn);
    expect(receivedGame.nameRu).toEqual(expectedGame.nameRu);
    expect(receivedGame.hasDemo).toEqual(expectedGame.hasDemo);
    expect(receivedGame.imageUrl).toEqual(expectedGame.imageUrl);
  });
}

export function checkGameInfoIsValid(game) {
  expect(game.id).not.toBeNil();
  expect(game.nameEn).not.toBeNil();
  expect(game.nameRu).toBeDefined();
  expect(game.hasDemo).not.toBeNil();
  expect(game.imageUrl).not.toBeNil();
}

export function checkJackpotValid(received) {
  expect(received.current.RUB).toBeGreaterThanOrEqual(0);
  expect(received.current.USD).toBeGreaterThanOrEqual(0);
  expect(received.current.EUR).toBeGreaterThanOrEqual(0);
  expect(received.current.UAH).toBeGreaterThanOrEqual(0);

  expect(received.current.RUB).not.toEqual(received.current.USD);
  expect(received.current.RUB).not.toEqual(received.current.EUR);
  expect(received.current.RUB).not.toEqual(received.current.UAH);
  expect(received.current.USD).not.toEqual(received.current.EUR);
  expect(received.current.USD).not.toEqual(received.current.UAH);
  expect(received.current.EUR).not.toEqual(received.current.UAH);

  expect(received.issued.RUB).toEqual(0);
  expect(received.issued.USD).toEqual(0);
  expect(received.issued.EUR).toEqual(0);
  expect(received.issued.UAH).toEqual(0);
}

export function compareJackpot(jackpot1, jackpot2, expectedValue) {
  if (expectedValue) {
    expect(jackpot1).toEqual(jackpot2);
  } else {
    expect(jackpot2.current.RUB).toBeGreaterThan(jackpot1.current.RUB);
    expect(jackpot2.current.USD).toBeGreaterThan(jackpot1.current.USD);
    expect(jackpot2.current.EUR).toBeGreaterThan(jackpot1.current.EUR);
    expect(jackpot2.current.UAH).toBeGreaterThan(jackpot1.current.UAH);
  }
}

export function checkPoker(data) {
  expect(data.gameHtml).toBeString();
  expect(data.gameScript).toBeNull();
  expect(data.gameHtml).toContain('id="cubeia-iframe"');
  expect(data.gameHtml).toMatch(/.*src=".*cubeia.com\/poker-client\/poker\/login\/200\/.*/);
}
