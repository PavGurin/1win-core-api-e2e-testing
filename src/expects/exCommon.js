export function checkTitles(data, expectedLang, expectedPath) {
  expect(data.success).toEqual(true);
  expect(data.titles.length).not.toEqual(0);
  data.titles.forEach((title) => {
    expect(title.id).toBeGreaterThan(0);
    if (expectedLang) {
      expect(title.lang).toEqual(expectedLang);
    } else {
      expect(title.lang).toBeString();
    }
    if (expectedPath) {
      expect(title.path).toEqual(expectedPath);
    } else {
      expect(title.path).toBeString();
    }
    expect(title.text).toBeString();
    expect(title.is_dynamic).toBeOneOf([0, 1]);
  });
}

export function checkEmptyTitles(data) {
  expect(data.success).toEqual(true);
  expect(data.titles.length).toEqual(0);
}
