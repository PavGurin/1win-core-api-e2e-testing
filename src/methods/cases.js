export const cases = {
  async getCaseInfo(caseId) {
    return socket.send(`CASES:cases-${caseId}`, {
    });
  },
  async getCaseCost(caseId) {
    const { data: { cost: caseCost } } = await socket.send(`CASES:cases-${caseId}`, {
    });
    return caseCost;
  },
  async playCaseWithoutChance(caseId) {
    return socket.send('CASES:cases-play', {
      id: caseId,
      chance: null,
      host: defaultVisitDomain,
    });
  },
  async raitingCase() {
    return socket.send('CASES:cases-rating', {});
  },
  async playCaseWithChance(caseId, chance) {
    return socket.send('CASES:cases-play', {
      id: caseId,
      chance,
      host: defaultVisitDomain,
    });
  },
  async getCaseInfoByTypeId(caseTypeId) {
    return socket.send('CASES:cases-casesByTypeId', { caseTypeId });
  },
};
