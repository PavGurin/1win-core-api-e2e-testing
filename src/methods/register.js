import {randomNum, randomStr} from '../randomizer';

const partner_key = 'test001';

export const register = {

    async one_click_reg() {
        return await socket.send('USER:auth-register',
            {
                isShort: true,
                country: default_country,
                timezone: 23,
                partner_key,
            });
    },

    async usual_reg(usualRegistration) {
        return await socket.send('USER:auth-register',
            {
                isShort: false,
                name: randomStr(),
                email: `${randomStr(5)}_test@new.xyz`,
                phone: randomNum().toString(),
                password: default_password,
                repeat_password: default_password,
                country: default_country,
                timezone: 23,
                birthday: 946587600000,
                partner_key,
                ...usualRegistration,
            });
    },
};
