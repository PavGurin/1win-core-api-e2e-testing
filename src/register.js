const partner_key = 'test001';
const birthday = 946587600002;
const default_password = '123456';

export const register = {

    async one_click_reg() {
        return await socket.send('USER:auth-register',
            {
                isShort: true,
                country: 'ru',
                timezone: 23,
                partner_key: partner_key
            });
    }
};

