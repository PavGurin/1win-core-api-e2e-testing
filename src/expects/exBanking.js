import { mysqlConnection } from '../methods/mysqlConnection';

export function successDepositCreate(data,
  expectedCurrency, expectedPaymentType, expectedAmount) {
  expect(data.apiResponse.error).toEqual(false);
  expect(data.currency).toEqual(expectedCurrency);
  expect(data.paymentType).toEqual(expectedPaymentType);
  expect(data.amount).toEqual(expectedAmount);
}

export async function getLastDeposit(userId) {
  return mysqlConnection.executeQuery(`SELECT * FROM 1win.ma_deposits
 WHERE id_user = ${userId} ORDER BY id DESC;`);
}

export async function successDbDeposit(userId,
  expectedAmount, expectedWallet, expectedType, expectedCurrency) {
  const dbItem = await getLastDeposit(userId);
  expect(dbItem[0].amount).toEqual(expectedAmount);
  expect(dbItem[0].currency).toEqual(expectedCurrency);
  expect(dbItem[0].payment_system).toEqual(expectedType);
  expect(dbItem[0].wallet).toEqual(expectedWallet);
}

export function successWithdrawalCreate(data) {
  expect(data).toBeObject();
  expect(data.email).not.toEqual(null);
  expect(data.message).toEqual(undefined);
}

export function checkUserdataSnippet(data, expectedPaymentMethod, expectedAmount) {
  expect(data.mostPaymentMethod).toEqual(expectedPaymentMethod);
  expect(data.paymentAmountSnippet).toEqual(expectedAmount);
}

export async function checkBalanceCreate(userId, currency) {
  const [res] = await mysqlConnection.executeQuery(`select * from 1win.ma_balance where id_user = '${userId}' and currency = '${currency}'`);
  expect(res).toBeDefined();
  expect(res.amount).toEqual(0);
  expect(res.power).toEqual(0);
  expect(res.percent).toEqual(0);
}
