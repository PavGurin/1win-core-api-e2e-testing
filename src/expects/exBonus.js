import { getUserBonusAmount } from '../methods/user';
import { banking } from '../methods/banking';

export async function checkBonus(userId, bet,
  expectedNewBonusAmount, expectedBetBonus, expectedBetProfitWithoutBonus) {
  expect(bet.status).toBe(2);
  expect(bet.profit.toFixed(2)).toBe(expectedBetProfitWithoutBonus.toFixed(2));
  expect(bet.promo_amount).toBe(expectedBetBonus);
  const newBonusAmount = await getUserBonusAmount(userId);
  expect(newBonusAmount).toBe(expectedNewBonusAmount.toString());
  const balance = await banking.balanceCheck();
  expect(balance.toFixed(2)).toEqual((expectedBetBonus + expectedBetProfitWithoutBonus).toFixed(2));
}

export async function checkExpressBonus(userId, bet,
  expectedNewBonusAmount, expectedBetBonus) {
  expect(bet.status).toBe(2);
  expect(bet.profit.toFixed(2)).not.toEqual('0.00');
  expect(bet.promo_amount).toBe(expectedBetBonus);
  const newBonusAmount = await getUserBonusAmount(userId);
  expect(newBonusAmount).toBe(expectedNewBonusAmount.toString());
  const balance = await banking.balanceCheck();
  expect(balance.toFixed(2)).toEqual((expectedBetBonus + bet.profit).toFixed(2));
}
