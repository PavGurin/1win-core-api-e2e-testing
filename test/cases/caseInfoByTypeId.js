import { cases } from '../../src/methods/cases';
import { register } from '../../src/methods/register';
import { checkCaseInfoByTypeId } from '../../src/expects/exCases';
import { getNewSocket } from '../global';

describe('Get cases by type id', () => {
  let socket;

  beforeAll(async () => {
    socket = await getNewSocket();
    await register.oneClickReg(socket);
  });

  it('C1461825 - TypeId = 1', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 1);
    checkCaseInfoByTypeId(data, 0, 1, 2);
    checkCaseInfoByTypeId(data, 1, 1, 11);
    checkCaseInfoByTypeId(data, 2, 1, 18);
  });

  it('C1461826 - TypeId = 2', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 2);
    checkCaseInfoByTypeId(data, 0, 2, 3);
    checkCaseInfoByTypeId(data, 1, 2, 10);
    checkCaseInfoByTypeId(data, 2, 2, 19);
  });

  it('C1461827 - TypeId = 3', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 3);
    checkCaseInfoByTypeId(data, 0, 3, 4);
    checkCaseInfoByTypeId(data, 1, 3, 12);
    checkCaseInfoByTypeId(data, 2, 3, 20);
  });

  it('C1461828 - TypeId = 4', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 4);
    checkCaseInfoByTypeId(data, 0, 4, 5);
    checkCaseInfoByTypeId(data, 1, 4, 13);
    checkCaseInfoByTypeId(data, 2, 4, 21);
  });

  it('C1461829 - TypeId = 5', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 5);
    checkCaseInfoByTypeId(data, 0, 5, 6);
    checkCaseInfoByTypeId(data, 1, 5, 22);
    checkCaseInfoByTypeId(data, 2, 5, 23);
  });

  it('C1461830 - TypeId = 6', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 6);
    checkCaseInfoByTypeId(data, 0, 6, 7);
    checkCaseInfoByTypeId(data, 1, 6, 24);
    checkCaseInfoByTypeId(data, 2, 6, 25);
  });

  it('C1461831 - TypeId = 7', async () => {
    const { data } = await cases.getCaseInfoByTypeId(socket, 7);
    checkCaseInfoByTypeId(data, 0, 7, 8);
    checkCaseInfoByTypeId(data, 1, 7, 26);
    checkCaseInfoByTypeId(data, 2, 7, 28);
  });
});
