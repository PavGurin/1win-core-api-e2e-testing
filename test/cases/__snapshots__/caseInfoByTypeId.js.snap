// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get cases by type id C1461825 - TypeId = 1 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "5",
      "20": "10",
      "30": "15",
    },
    "caseTypeId": 1,
    "cost": 20,
    "currency": "RUB",
    "id": 2,
    "max": 200,
    "min": 5,
    "name": "Везунчик",
    "priceMaxWin": 200,
    "priceMinWin": 5,
    "totallyWon": 35665,
  },
  Object {
    "adjust": Object {
      "10": "0.08",
      "20": "0.2",
      "30": "0.25",
    },
    "caseTypeId": 1,
    "cost": 0.4,
    "currency": "USD",
    "id": 11,
    "max": 4,
    "min": 0.08,
    "name": "Везунчик",
    "priceMaxWin": 4,
    "priceMinWin": 0.08,
    "totallyWon": 100839.52,
  },
  Object {
    "adjust": Object {
      "10": "0.08",
      "20": "0.2",
      "30": "0.25",
    },
    "caseTypeId": 1,
    "cost": 0.4,
    "currency": "EUR",
    "id": 18,
    "max": 4,
    "min": 0.08,
    "name": "Везунчик",
    "priceMaxWin": 4,
    "priceMinWin": 0.08,
    "totallyWon": 100143.15,
  },
]
`;

exports[`Get cases by type id C1461826 - TypeId = 2 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "10",
      "20": "30",
      "30": "50",
    },
    "caseTypeId": 2,
    "cost": 50,
    "currency": "RUB",
    "id": 3,
    "max": 500,
    "min": 10,
    "name": "Топовый",
    "priceMaxWin": 500,
    "priceMinWin": 10,
    "totallyWon": 86568,
  },
  Object {
    "adjust": Object {
      "10": "0.2",
      "20": "0.4",
      "30": "0.5",
    },
    "caseTypeId": 2,
    "cost": 1,
    "currency": "USD",
    "id": 10,
    "max": 10,
    "min": 0.2,
    "name": "Топовый",
    "priceMaxWin": 10,
    "priceMinWin": 0.2,
    "totallyWon": 252071.99,
  },
  Object {
    "adjust": Object {
      "10": "0.2",
      "20": "0.4",
      "30": "0.5",
    },
    "caseTypeId": 2,
    "cost": 1,
    "currency": "EUR",
    "id": 19,
    "max": 10,
    "min": 0.2,
    "name": "Топовый",
    "priceMaxWin": 10,
    "priceMinWin": 0.2,
    "totallyWon": 251361.23,
  },
]
`;

exports[`Get cases by type id C1461827 - TypeId = 3 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "50",
      "20": "150",
      "30": "250",
    },
    "caseTypeId": 3,
    "cost": 100,
    "currency": "RUB",
    "id": 4,
    "max": 500,
    "min": 50,
    "name": "Фартовый",
    "priceMaxWin": 500,
    "priceMinWin": 50,
    "totallyWon": 391564,
  },
  Object {
    "adjust": Object {
      "10": "0.2",
      "20": "0.6",
      "30": "1",
    },
    "caseTypeId": 3,
    "cost": 2,
    "currency": "USD",
    "id": 12,
    "max": 10,
    "min": 1,
    "name": "Фартовый",
    "priceMaxWin": 10,
    "priceMinWin": 1,
    "totallyWon": 28671.17,
  },
  Object {
    "adjust": Object {
      "10": "0.2",
      "20": "0.6",
      "30": "1",
    },
    "caseTypeId": 3,
    "cost": 2,
    "currency": "EUR",
    "id": 20,
    "max": 10,
    "min": 1,
    "name": "Фартовый",
    "priceMaxWin": 10,
    "priceMinWin": 1,
    "totallyWon": 26165.88,
  },
]
`;

exports[`Get cases by type id C1461828 - TypeId = 4 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "2",
      "20": "3",
      "30": "5",
    },
    "caseTypeId": 4,
    "cost": 500,
    "currency": "RUB",
    "id": 5,
    "max": 5000,
    "min": 100,
    "name": "Лакшери",
    "priceMaxWin": 5000,
    "priceMinWin": 100,
    "totallyWon": 495140,
  },
  Object {
    "adjust": Object {
      "10": "0.2",
      "20": "0.6",
      "30": "1",
    },
    "caseTypeId": 4,
    "cost": 10,
    "currency": "USD",
    "id": 13,
    "max": 100,
    "min": 2,
    "name": "Лакшери",
    "priceMaxWin": 100,
    "priceMinWin": 2,
    "totallyWon": 59339.15,
  },
  Object {
    "adjust": Object {
      "10": "0.2",
      "20": "0.6",
      "30": "1",
    },
    "caseTypeId": 4,
    "cost": 10,
    "currency": "EUR",
    "id": 21,
    "max": 100,
    "min": 2,
    "name": "Лакшери",
    "priceMaxWin": 100,
    "priceMinWin": 2,
    "totallyWon": 53300.92,
  },
]
`;

exports[`Get cases by type id C1461829 - TypeId = 5 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "100",
      "20": "250",
      "30": "500",
    },
    "caseTypeId": 5,
    "cost": 1000,
    "currency": "RUB",
    "id": 6,
    "max": 10000,
    "min": 500,
    "name": "Олигарх",
    "priceMaxWin": 10000,
    "priceMinWin": 500,
    "totallyWon": 1660362,
  },
  Object {
    "adjust": Object {
      "10": "2",
      "20": "5",
      "30": "10",
    },
    "caseTypeId": 5,
    "cost": 20,
    "currency": "USD",
    "id": 22,
    "max": 200,
    "min": 10,
    "name": "Олигарх",
    "priceMaxWin": 200,
    "priceMinWin": 10,
    "totallyWon": 5323.22,
  },
  Object {
    "adjust": Object {
      "10": "2",
      "20": "5",
      "30": "10",
    },
    "caseTypeId": 5,
    "cost": 20,
    "currency": "EUR",
    "id": 23,
    "max": 200,
    "min": 10,
    "name": "Олигарх",
    "priceMaxWin": 200,
    "priceMinWin": 10,
    "totallyWon": 93.23,
  },
]
`;

exports[`Get cases by type id C1461830 - TypeId = 6 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "500",
      "20": "1500",
      "30": "2500",
    },
    "caseTypeId": 6,
    "cost": 5000,
    "currency": "RUB",
    "id": 7,
    "max": 10000,
    "min": 2500,
    "name": "Миллионер",
    "priceMaxWin": 10000,
    "priceMinWin": 2500,
    "totallyWon": 4098814,
  },
  Object {
    "adjust": Object {
      "10": "10",
      "20": "30",
      "30": "50",
    },
    "caseTypeId": 6,
    "cost": 100,
    "currency": "USD",
    "id": 24,
    "max": 200,
    "min": 50,
    "name": "Миллионер",
    "priceMaxWin": 200,
    "priceMinWin": 50,
    "totallyWon": 19887.84,
  },
  Object {
    "adjust": Object {
      "10": "10",
      "20": "30",
      "30": "50",
    },
    "caseTypeId": 6,
    "cost": 100,
    "currency": "EUR",
    "id": 25,
    "max": 200,
    "min": 50,
    "name": "Миллионер",
    "priceMaxWin": 200,
    "priceMinWin": 50,
    "totallyWon": 730.17,
  },
]
`;

exports[`Get cases by type id C1461831 - TypeId = 7 1`] = `
Array [
  Object {
    "adjust": Object {
      "10": "1000",
      "20": "2500",
      "30": "5000",
    },
    "caseTypeId": 7,
    "cost": 10000,
    "currency": "RUB",
    "id": 8,
    "max": 50000,
    "min": 5000,
    "name": "Хозяин жизни",
    "priceMaxWin": 50000,
    "priceMinWin": 5000,
    "totallyWon": 12474140,
  },
  Object {
    "adjust": Object {
      "10": "20",
      "20": "60",
      "30": "100",
    },
    "caseTypeId": 7,
    "cost": 200,
    "currency": "USD",
    "id": 26,
    "max": 1000,
    "min": 100,
    "name": "Хозяин жизни",
    "priceMaxWin": 1000,
    "priceMinWin": 100,
    "totallyWon": 9349235.76,
  },
  Object {
    "adjust": Object {
      "10": "20",
      "20": "60",
      "30": "100",
    },
    "caseTypeId": 7,
    "cost": 200,
    "currency": "EUR",
    "id": 28,
    "max": 1000,
    "min": 100,
    "name": "Хозяин жизни",
    "priceMaxWin": 1000,
    "priceMinWin": 100,
    "totallyWon": 9349636.53,
  },
]
`;
